packages/react/src/events.ts

```diff
@@ -3,21 +3,29 @@ import {
   HandlerFunction,
   isEventDataState,
   ServerEventDataState,
+  ServerHandlerDataState,
   StateDataState,
 } from './template'
 
 export function isServerEventDataState(obj: any): obj is ServerEventDataState {
   return isEventDataState(obj) && 'handler' in obj && typeof obj.handler === 'function'
 }
 
-export function event<T>(
+export function event<T extends any[]>(
   func: HandlerFunction<T>,
+  opts?: {
+    actions?: ActionDataState[]
+    timeout?: number
+  }
+): ServerEventDataState<T>
+export function event<T>(
+  func: HandlerFunction<[T]>,
   opts?: {
     actions?: ActionDataState[]
     timeout?: number
     args?: { [K in keyof T]: StateDataState<T[K]> }
   }
-): ServerEventDataState {
+): ServerEventDataState<[T]> {
```

Comment:
When you specify "args", payload is overwritten and you receive just one argument - your state values (as an object).